platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/CacheBuilder.java

```diff
@@ -59,8 +59,18 @@ enum LockTarget {
 
     /**
      * Specifies the <em>initial</em> lock options to use. See {@link PersistentCache} for details.
-     *
+     * <br>
      * Note that not every combination of cache type and lock options is supported.
+     * <br>
+     * Supported LockModes:
+     * <br>- Shared: Default option. Used for read-only caches, many processes can access the cache concurrently.
+     *      The cache lock is created when cache is opened. To release a lock a cache has to be closed.
+     * <br>- Exclusive: Used for caches that are written to. Only one process can access the cache at a time.
+     *      The cache lock is created when cache is opened. To release a lock a cache has to be closed.
+     * <br>- OnDemand: Used for caches that are written to on demand. Only one process can access the cache at a time.
+     *      The cache lock is created with `useCache` or `withFiles` method. Lock is released when another process requests it via ping requests,
+     *      see {@link org.gradle.cache.internal.DefaultFileLockManager.DefaultFileLock#lock(org.gradle.cache.FileLockManager.LockMode)}.
+     * <br>- None: No locking whatsoever.
```

Comment:
I think this should go to the `LockMode` type.

Reply:
I thought that too, but this behaviour is specific to PersistentCache that is returned from a build. While you can use LockMode without  PersistentCache and behaviour that might be different (e.g. OnDemand could be implemented differently).

Reply:
Hm. That is unfortunate. Do we ever do that, i.e. use `LockMode` outside of `PersistentCache`? Can we somehow reduce the complexity here? Can we only use `LockMode` via `PersistentCache`? If not, can we make the other use cases work the same way as with `PersistentCache`? If not, then I think we should not reuse the same `LockMode` type and instead define a separate type.

Reply:
Yes we do, where we use FileLockManager directly, e.g. in toolchains [JdkCacheDirectory](https://github.com/gradle/gradle/blob/b2deab3a2e6171cddcf143090f32ee326bd1b5ab/platforms/jvm/toolchains-jvm/src/main/java/org/gradle/jvm/toolchain/internal/install/JdkCacheDirectory.java#L281) and in [DefaultBuildLogicBuildQueue](https://github.com/gradle/gradle/blob/b2deab3a2e6171cddcf143090f32ee326bd1b5ab/subprojects/core/src/main/java/org/gradle/api/internal/initialization/DefaultBuildLogicBuildQueue.java#L93) and for cc instrumentation. 

I think there are some cases where you want to use just FileLockManager and not other caching infrastructure.
I think creating something like CacheLockOptions and CacheLockMode that then gets translated to LockOptions and LockMode makes sense.

Reply:
It requires some code replacement, especially in tests

Reply:
Alternatively we can move comments to the `LockMode` and we write what is a behaviour if some option is used with the `PersistentCache`, and if it's used with FileLockerManager